cedar-policy-core/src/entities/json/err.rs

```diff
@@ -14,72 +14,97 @@
-    ExprParseError(crate::parser::err::WithContext),
-    /// Contents of an `__entity` escape failed to parse as an entity reference
-    #[error(transparent)]
-    EntityParseError(crate::parser::err::WithContext),
-    /// Function name in an `__extn` escape failed to parse as an extension function name
-    #[error(transparent)]
-    ExtnParseError(crate::parser::err::WithContext),
+    /// Note that `__expr` is deprecated and once it is
+    /// removed, this error will also be removed
+    #[error("failed to parse escape `{kind}`: {value}, errors: {errs}")]
+    ParseEscape {
+        /// Escape kind
+        kind: EscapeKind,
+        /// Escape value at fault
+        value: String,
+        /// Parse errors
+        errs: ParseErrors,
+    },
     /// Restricted expression error
     #[error(transparent)]
     RestrictedExpressionError(#[from] RestrictedExpressionError),
     /// Error thrown by an operation on `Extensions`
     #[error(transparent)]
     ExtensionsError(#[from] ExtensionsError),
     /// A field that needs to be a literal entity reference, was some other JSON value
-    #[error("{ctx}, expected a literal entity reference, but got {got}")]
+    #[error("{ctx}, expected a literal entity reference, but got `{got}`")]
     ExpectedLiteralEntityRef {
         /// Context of this error
         ctx: Box<JsonDeserializationErrorContext>,
         /// the expression we got instead
         got: Box<Expr>,
     },
     /// A field that needs to be an extension value, was some other JSON value
-    #[error("{ctx}, expected an extension value, but got {got}")]
+    #[error("{ctx}, expected an extension value, but got `{got}`")]
     ExpectedExtnValue {
         /// Context of this error
         ctx: Box<JsonDeserializationErrorContext>,
         /// the expression we got instead
         got: Box<Expr>,
     },
     /// Contexts need to be records, but we got some other JSON value
-    #[error("Expected Context to be a record, but got {got}")]
+    #[error("expected `context` to be a record, but got `{got}`")]
     ExpectedContextToBeRecord {
         /// Expression we got instead
         got: Box<RestrictedExpr>,
     },
     /// Parents of actions should be actions, but this action has a non-action parent
-    #[error("{uid} is an action, so it should not have a parent {parent}, which is not an action")]
-    ActionParentIsNonAction {
+    #[error("action `{uid}` has a non-action parent `{parent}`")]
+    ActionParentIsNotAction {
         /// Action entity that had the invalid parent
         uid: EntityUID,
         /// Parent that is invalid
         parent: EntityUID,
     },
     /// Schema-based parsing needed an implicit extension constructor, but no suitable
     /// constructor was found
-    #[error("{ctx}, extension constructor for {arg_type} -> {return_type} not found")]
-    ImpliedConstructorNotFound {
+    #[error("{ctx}, missing extension constructor for `{arg_type}` -> `{return_type}`")]
```

Comment:
We don't need backticks around the types here, as they are not user-provided strings
```suggestion
    #[error("{ctx}, missing extension constructor for {arg_type} -> {return_type}")]
```